### PR TITLE
fix(35779): Comments at the end of an array, when the last item ends with a comma, are not compiled

### LINKED ISSUE #35779: Comments at the end of an array, when the last item ends with a comma, are not compiled
**TypeScript Version:**  3.7.2

**Search Terms:** comments array end

**Expected behavior:**  All comments should be compiled to javacript

**Actual behavior:** Comments at the end of an array, when the last item ends with a comma, are not compiled. 

**Related Issues:** #5810

**Code**
```ts
const a = [
    //comment is in output
    1,
    2,
    //comment disappears
]
```

<details><summary><b>Output</b></summary>

```ts
"use strict";
const a = [
    //comment is in output
    1,
    2,
];

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/?ssl=5&ssc=7&pln=5&pc=13#code/MYewdgzgLgBAhjAvDA2gKBpmB6boC2+ApmLAJYQxlgwgCuUADgxlgIwA0rmATF1jjwhCJWABMKcRoyJwAThDQBdIA)

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #35779

### CODE REVIEW

**Comment by a-tarasyuk** [comment]
@weswigham oke, I'll check. However, I think `emitLeadingCommentsOfPosition` won't work for the following case, because the comment is not _leading_. Anyway, I need to check.  
```ts
const a = [1, /* comment */];
```

**Comment by a-tarasyuk** [comment]
Just noticed that comments removed from object literal too, 

```ts
// a.ts
const a = {
  x: 1, // comment
  y: 1, /* comment */
  z: 1, /* comment */
}

// a.js
"use strict";
const a = {
    x: 1,
    y: 1,
    z: 1,
};

```

Is it expected?

**Comment by weswigham** [comment]
It'd be nice if this included the test from the original issue,
```ts
const a = [
    //comment is in output
    1,
    2,
    //comment disappears
]
```
but it's probably not strictly needed, since it looks like another one of our tests happens to have a similar comment which is now preserved. I'll let @rbuckton have the final say, as the second reviewer (he can merge when he's done).

File: tests/baselines/reference/commentOnArrayElement16.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+];
+
+
+//// [commentOnArrayElement16.js]
+var array = [
+    // comment start
+    1,
+    2,
+    // comment end
+];
```
Comment (a-tarasyuk) [inline tests/baselines/reference/commentOnArrayElement16.js:16]: @weswigham I added the test from the original issue :)

**Pull request merged**